lib/shelley/src/Cardano/Wallet/Shelley/Api/Server.hs

```diff
@@ -0,0 +1,256 @@
+
+import Prelude
+
+import Cardano.Wallet
+    ( ErrCreateRandomAddress (..)
+    , ErrValidateSelection
+    , genesisData
+    , networkLayer
+    )
+import Cardano.Wallet.Api
+    ( Addresses
+    , Api
+    , ApiLayer (..)
+    , ByronAddresses
+    , ByronMigrations
+    , ByronTransactions
+    , ByronWallets
+    , CoinSelections
+    , Network
+    , Proxy_
+    , StakePools
+    , Transactions
+    , Wallets
+    )
+import Cardano.Wallet.Api.Server
+    ( deleteTransaction
+    , deleteWallet
+    , getMigrationInfo
+    , getNetworkClock
+    , getNetworkInformation
+    , getNetworkParameters
+    , getUTxOsStatistics
+    , getWallet
+    , liftHandler
+    , listAddresses
+    , listTransactions
+    , listWallets
+    , mkLegacyWallet
+    , postExternalTransaction
+    , postIcarusWallet
+    , postLedgerWallet
+    , postRandomAddress
+    , postRandomWallet
+    , postRandomWalletFromXPrv
+    , postTransaction
+    , postTransactionFee
+    , postTrezorWallet
+    , putByronWalletPassphrase
+    , putWallet
+    , rndStateChange
+    , withLegacyLayer
+    , withLegacyLayer'
+    )
+import Cardano.Wallet.Api.Types
+    ( ApiT (..), SomeByronWalletPostData (..) )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( PaymentAddress (..) )
+import Cardano.Wallet.Primitive.AddressDerivation.Byron
+    ( ByronKey )
+import Cardano.Wallet.Primitive.AddressDerivation.Icarus
+    ( IcarusKey )
+import Cardano.Wallet.Primitive.AddressDiscovery.Random
+    ( RndState )
+import Cardano.Wallet.Primitive.AddressDiscovery.Sequential
+    ( SeqState )
+import Control.Applicative
+    ( liftA2 )
+import Control.Monad.Trans.Except
+    ( throwE )
+import Data.Coerce
+    ( coerce )
+import Data.Generics.Internal.VL.Lens
+    ( (^.) )
+import Data.Generics.Labels
+    ()
+import Data.List
+    ( sortOn )
+import Fmt
+    ( Buildable )
+import Network.Ntp
+    ( NtpClient )
+import Servant
+    ( (:<|>) (..), Server, err501, throwError )
+
+-- | A diminished servant server to serve Byron wallets only.
+server
+    :: forall t n.
+        ( Buildable (ErrValidateSelection t)
+        , PaymentAddress n IcarusKey
+        , PaymentAddress n ByronKey
+        )
+    => ApiLayer (SeqState n IcarusKey) t IcarusKey
+    -> NtpClient
+    -> Server (Api n)
+server icarus ntp =
+         wallets
+    :<|> addresses
+    :<|> coinSelections
+    :<|> transactions
+    :<|> stakePools
+    :<|> byronWallets
+    :<|> byronAddresses
+    :<|> byronTransactions
+    :<|> byronMigrations
+    :<|> network
+    :<|> proxy
+  where
+    byron :: ApiLayer (RndState n) t ByronKey
+    byron = error "todo: should we have a byron api here?"
+
+    wallets :: Server Wallets
+    wallets =
+             (\_ -> throwError err501)
+        :<|> (\_ -> throwError err501)
+        :<|> throwError err501
+        :<|> (\_ -> throwError err501)
+        :<|> (\_ _ -> throwError err501)
+        :<|> (\_ _ -> throwError err501)
+        :<|> (\_ -> throwError err501)
```

Comment:
These will now be accessible in Shelley.

Reply:
The shelley wallet endpoints are now added.